## Part 1

2. Implementation

In [48]:
import pandas as pd
import numpy as np
import sklearn as sk
from IPython.display import display
from sklearn import preprocessing


In [49]:
%matplotlib inline
data = pd.read_csv("data.csv")
(data.head())

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2728: DtypeWarning: Columns (0,4,5,6,7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


Id                                              Title  \
0  37404348                           Casual Stock Replenisher   
1  37404337                           Casual Stock Replenisher   
2  37404356  RETAIL SALES SUPERSTARS and STYLISTS Wanted - ...   
3  37404330                              Team member - Belrose   
4  37404308  Business Banking Contact Centre Specialist, Ni...   

                                          Company                      Date  \
0                                     Aldi Stores  2018-10-07T00:00:00.000Z   
1                                     Aldi Stores  2018-10-07T00:00:00.000Z   
2                             LB Creative Pty Ltd  2018-10-07T00:00:00.000Z   
3                          Anaconda Group Pty Ltd  2018-10-07T00:00:00.000Z   
4  Commonwealth Bank - Business & Private Banking  2018-10-07T00:00:00.000Z   

                  Location                         Area  \
0                   Sydney  North West & Hills District   
1    Richmond & Hawkesbury                          NaN   
2                 Brisbane          CBD & Inner Suburbs   
3  Gosford & Central Coast                          NaN   
4                   Sydney        Ryde & Macquarie Park   

                   Classification  SubClassification  \
0      Retail & Consumer Products  Retail Assistants   
1      Retail & Consumer Products  Retail Assistants   
2      Retail & Consumer Products  Retail Assistants   
3      Retail & Consumer Products  Retail Assistants   
4  Call Centre & Customer Service    Sales - Inbound   

                                         Requirement FullDescription  \
0  Our Casual Stock Replenishers pride themselves...             NaN   
1  Our Casual Stock Replenishers pride themselves...             NaN   
2  BRAND NEW FLAGSHIP STORE OPENING - SUNSHINE PLAZA             NaN   
3  Bring it on - do you love the great outdoors a...             NaN   
4  We are seeking highly articulate, enthusiastic...             NaN   

   LowestSalary  HighestSalary JobType  
0             0             30     NaN  
1             0             30     NaN  
2             0             30     NaN  
3             0             30     NaN  
4             0             30     NaN

## 1. Describe the dataset

## Categories/domains

In [50]:
list(data)

['Id',
 'Title',
 'Company',
 'Date',
 'Location',
 'Area',
 'Classification',
 'SubClassification',
 'Requirement',
 'FullDescription',
 'LowestSalary',
 'HighestSalary',
 'JobType']

## Size of each variation (318477)

In [51]:
data.shape

(318477, 13)

## Structure/format

In [52]:
data.dtypes

Id                   object
Title                object
Company              object
Date                 object
Location             object
Area                 object
Classification       object
SubClassification    object
Requirement          object
FullDescription      object
LowestSalary          int64
HighestSalary         int64
JobType              object
dtype: object

In [53]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 318477 entries, 0 to 318476
Data columns (total 13 columns):
Id                   318477 non-null object
Title                318477 non-null object
Company              306473 non-null object
Date                 318477 non-null object
Location             197229 non-null object
Area                 122658 non-null object
Classification       197229 non-null object
SubClassification    197229 non-null object
Requirement          318470 non-null object
FullDescription      302302 non-null object
LowestSalary         318477 non-null int64
HighestSalary        318477 non-null int64
JobType              302379 non-null object
dtypes: int64(2), object(11)
memory usage: 31.6+ MB


## attributes/features of review data 

For the purpost of this analysis, we are going to need to look at all the countries except for id

This is because we will be examining how the job market at different location changes over time. This period over which the data were collected rages from 2018-10-01 to 2010-03-13

Other features such as job classification and requirements will also be used to support out analysis

Lowest and Highest Salary is going to tell us the relationship between job classicication and the amount of pay

## 2.Steps used for data preparation and preprocessing 

Load data

In [54]:
data = pd.read_csv("data.csv")

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2728: DtypeWarning: Columns (0,4,5,6,7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


Normalize the data

In [55]:
data_norm = (data["LowestSalary"] - data["LowestSalary"].min()) / (data["LowestSalary"].max() - data["LowestSalary"].min())*20
data_norm

0          0.0
1          0.0
2          0.0
3          0.0
4          0.0
5          0.0
6          0.0
7          0.0
8          0.0
9          0.0
10         0.0
11         0.0
12         0.0
13         0.0
14         0.0
15         0.0
16         0.0
17         0.0
18         0.0
19         0.0
20         0.0
21         0.0
22         0.0
23         0.0
24         0.0
25         0.0
26         0.0
27         0.0
28         0.0
29         0.0
          ... 
318447    12.0
318448    12.0
318449    12.0
318450    12.0
318451    12.0
318452    12.0
318453    12.0
318454    12.0
318455    12.0
318456    12.0
318457    12.0
318458    15.0
318459    15.0
318460    15.0
318461    15.0
318462    15.0
318463    15.0
318464    15.0
318465    15.0
318466    15.0
318467    15.0
318468    15.0
318469    20.0
318470    20.0
318471    20.0
318472    20.0
318473    20.0
318474    20.0
318475    20.0
318476    20.0
Name: LowestSalary, Length: 318477, dtype: float64

In [56]:
data_norm = (data["HighestSalary"] - data["HighestSalary"].min()) / (data["HighestSalary"].max() - data["HighestSalary"].min())*20
data_norm

0          0.000000
1          0.000000
2          0.000000
3          0.000000
4          0.000000
5          0.000000
6          0.000000
7          0.000000
8          0.000000
9          0.000000
10         0.000000
11         0.000000
12         0.000000
13         0.000000
14         0.000000
15         0.000000
16         0.000000
17         0.000000
18         0.000000
19         0.000000
20         0.000000
21         0.000000
22         0.000000
23         0.000000
24         0.000000
25         0.000000
26         0.000000
27         0.000000
28         0.000000
29         0.000000
            ...    
318447     2.476780
318448     2.476780
318449     2.476780
318450     2.476780
318451     2.476780
318452     2.476780
318453     2.476780
318454     2.476780
318455     2.476780
318456     2.476780
318457     2.476780
318458     3.508772
318459     3.508772
318460     3.508772
318461     3.508772
318462     3.508772
318463     3.508772
318464     3.508772
318465     3.508772


In [57]:
data['JobType'].unique()

array([nan, 'Full Time', 'Contract/Temp', 'Part Time', 'Casual/Vacation'],
      dtype=object)

In [58]:
data['Area'].isnull().sum()

195819

In [59]:
data['Location'].isnull().sum()

121248

In [60]:
data['Classification'].isnull().sum()

121248

In [61]:
data['SubClassification'].isnull().sum()

121248

We can see that there are a lot of missing values for Location, Area, Classification, and sub-Classification but we are not sure if the analysis need to focus on this part

## Clean the data

In [62]:
data.drop(['Id'],axis=1,inplace=True)

In [63]:
data=data.dropna(axis=0, how ='all')

In [64]:
data.tail()

Title  \
318472                 Program Manager - Trade Reporting   
318473           Web Content Writer - June 2019 contract   
318474             Brand Director - Global Premium Brand   
318475  Head of Financial Planning, Reporting & Strategy   
318476                          Head of Operations - EOI   

                                                  Company  \
318472                                            Talenza   
318473  Talent – Winner ‘Seek Large Recruitment Agency...   
318474                   Retail Career Consulting Pty Ltd   
318475                                         Moir Group   
318476                                 Austcorp Executive   

                            Date Location Area Classification  \
318472  2019-03-13T00:00:00.000Z      NaN  NaN            NaN   
318473  2019-03-13T00:00:00.000Z      NaN  NaN            NaN   
318474  2019-03-12T00:00:00.000Z      NaN  NaN            NaN   
318475  2019-03-08T00:00:00.000Z      NaN  NaN            NaN   
318476  2019-03-13T00:00:00.000Z      NaN  NaN            NaN   

       SubClassification                                        Requirement  \
318472               NaN  Our client is a large Financial Services firm ...   
318473               NaN  You will be able to write clearly and concisel...   
318474               NaN  Newly created Brand Director AUS. Must be comm...   
318475               NaN  High-quality manufacturing organisation. Excit...   
318476               NaN      Expressions of interest for a leadership role   

       FullDescription  LowestSalary  HighestSalary JobType  
318472             NaN           200            999     NaN  
318473             NaN           200            999     NaN  
318474             NaN           200            999     NaN  
318475             NaN           200            999     NaN  
318476             NaN           200            999     NaN

## Hypothesis about the analysis outcome?

If we compare average salary per location, we think that bigger cities will have a higher salary (looking at variation of salary within that city)
Also we think that there will be mom demans for job in the main citites
If there a more demand for a certain type of job classication, salary would be higher

## PART 2

## 1) Extract the relevant information to describe the job's attributes

## 2) Study the market by locations(Vu)

In [65]:
data.head()

Title  \
0                           Casual Stock Replenisher   
1                           Casual Stock Replenisher   
2  RETAIL SALES SUPERSTARS and STYLISTS Wanted - ...   
3                              Team member - Belrose   
4  Business Banking Contact Centre Specialist, Ni...   

                                          Company                      Date  \
0                                     Aldi Stores  2018-10-07T00:00:00.000Z   
1                                     Aldi Stores  2018-10-07T00:00:00.000Z   
2                             LB Creative Pty Ltd  2018-10-07T00:00:00.000Z   
3                          Anaconda Group Pty Ltd  2018-10-07T00:00:00.000Z   
4  Commonwealth Bank - Business & Private Banking  2018-10-07T00:00:00.000Z   

                  Location                         Area  \
0                   Sydney  North West & Hills District   
1    Richmond & Hawkesbury                          NaN   
2                 Brisbane          CBD & Inner Suburbs   
3  Gosford & Central Coast                          NaN   
4                   Sydney        Ryde & Macquarie Park   

                   Classification  SubClassification  \
0      Retail & Consumer Products  Retail Assistants   
1      Retail & Consumer Products  Retail Assistants   
2      Retail & Consumer Products  Retail Assistants   
3      Retail & Consumer Products  Retail Assistants   
4  Call Centre & Customer Service    Sales - Inbound   

                                         Requirement FullDescription  \
0  Our Casual Stock Replenishers pride themselves...             NaN   
1  Our Casual Stock Replenishers pride themselves...             NaN   
2  BRAND NEW FLAGSHIP STORE OPENING - SUNSHINE PLAZA             NaN   
3  Bring it on - do you love the great outdoors a...             NaN   
4  We are seeking highly articulate, enthusiastic...             NaN   

   LowestSalary  HighestSalary JobType  
0             0             30     NaN  
1             0             30     NaN  
2             0             30     NaN  
3             0             30     NaN  
4             0             30     NaN

# Number of different job sectors in each city

In [66]:
jobNum = data.groupby('Location')['SubClassification'].nunique()
jobNum

Location
ACT                                     277
Adelaide                                298
Adelaide Hills & Barossa                 67
Albany & Great Southern                  63
Albury Area                             126
Alice Springs & Central Australia        83
Bairnsdale & Gippsland                  110
Ballarat & Central Highlands            144
Bendigo, Goldfields & Macedon Ranges    152
Blue Mountains & Central West           169
Brisbane                                329
Broome & Kimberley                       84
Bunbury & South West                    128
Bundaberg & Wide Bay Burnett            105
Cairns & Far North                      196
Central & South East TAS                 32
Coffs Harbour & North Coast             117
Coober Pedy & Outback SA                 40
Darwin                                  195
Devonport & North West                   98
Dubbo & Central NSW                     134
Far West & North Central NSW             64
Fleurieu Peninsula & Ka

# Number of jobs (market size) in each city

In [67]:
marketSize = data.groupby('Location').size()
marketSize

Location
ACT                                      6250
Adelaide                                 6653
Adelaide Hills & Barossa                  193
Albany & Great Southern                   150
Albury Area                               416
Alice Springs & Central Australia         301
Bairnsdale & Gippsland                    378
Ballarat & Central Highlands              603
Bendigo, Goldfields & Macedon Ranges      808
Blue Mountains & Central West             942
Brisbane                                21466
Broome & Kimberley                        283
Bunbury & South West                      635
Bundaberg & Wide Bay Burnett              409
Cairns & Far North                       1509
Central & South East TAS                   57
Coffs Harbour & North Coast               440
Coober Pedy & Outback SA                  112
Darwin                                   1266
Devonport & North West                    256
Dubbo & Central NSW                       642
Far West & North Central 

# Hottest job sectors in each city

In [68]:
hottestJob = data.groupby('Location')['SubClassification'].max()
hottestJob

Location
ACT                                         Workplace Training & Assessment
Adelaide                                    Workplace Training & Assessment
Adelaide Hills & Barossa                               Winery & Viticulture
Albany & Great Southern                              Welders & Boilermakers
Albury Area                                 Workplace Training & Assessment
Alice Springs & Central Australia           Workplace Training & Assessment
Bairnsdale & Gippsland                               Welders & Boilermakers
Ballarat & Central Highlands                Workplace Training & Assessment
Bendigo, Goldfields & Macedon Ranges                 Welders & Boilermakers
Blue Mountains & Central West                        Welders & Boilermakers
Brisbane                                    Workplace Training & Assessment
Broome & Kimberley                          Workplace Training & Assessment
Bunbury & South West                        Workplace Training & Assessment
Bun

# The common salary range in each city

In [69]:
salaryRangeLoc = data.groupby('Location')['LowestSalary', 'HighestSalary'].mean()
salaryRangeLoc

LowestSalary  HighestSalary
Location                                                         
ACT                                      86.174400     198.718240
Adelaide                                 57.991883      97.021494
Adelaide Hills & Barossa                 41.191710      68.948187
Albany & Great Southern                  56.600000     101.233333
Albury Area                              51.201923      75.543269
Alice Springs & Central Australia        64.186047     113.647841
Bairnsdale & Gippsland                   55.740741     105.328042
Ballarat & Central Highlands             53.067993      84.243781
Bendigo, Goldfields & Macedon Ranges     49.504950      70.402228
Blue Mountains & Central West            60.764331     108.487261
Brisbane                                 65.875804     121.621588
Broome & Kimberley                       66.431095     117.628975
Bunbury & South West                     50.866142      79.795276
Bundaberg & Wide Bay Burnett             61.149144     127.176039
Cairns & Far North                       54.440027      94.074884
Central & South East TAS                 51.754386      68.245614
Coffs Harbour & North Coast              54.318182     101.870455
Coober Pedy & Outback SA                 73.303571     107.125000
Darwin                                   68.467615     120.466035
Devonport & North West                   53.007812      87.750000
Dubbo & Central NSW                      66.915888     116.037383
Far West & North Central NSW             69.289340     110.888325
Fleurieu Peninsula & Kangaroo Island     42.968750      58.125000
Geraldton, Gascoyne & Midwest            73.489736     128.756598
Gladstone & Central QLD                  66.502809     104.528090
Gold Coast                               49.721293      82.994426
Gosford & Central Coast                  54.296351      88.876396
Hervey Bay & Fraser Coast                53.263158      99.575439
Hobart                                   55.480495      94.739296
Horsham & Grampians                      56.937799      91.698565
...                                            ...            ...
Mandurah & Peel                          56.320000      90.140000
Melbourne                                64.252302     113.982716
Mildura & Murray                         60.586319     118.872964
Mornington Peninsula & Bass Coast        43.951286      73.446549
Mt Gambier & Limestone Coast             55.178571     104.601190
Newcastle, Maitland & Hunter             59.044972     102.482062
Northam & Wheatbelt                      50.500000      77.362500
Northern QLD                             59.792364     113.129940
Perth                                    66.969238     117.191090
Port Hedland, Karratha & Pilbara         77.522496     121.421716
Port Macquarie & Mid North Coast         71.812367     149.605544
Richmond & Hawkesbury                    49.000000      77.842857
Riverland & Murray Mallee                48.926174      64.697987
Rockhampton & Capricorn Coast            64.972527     114.673077
Shepparton & Goulburn Valley             51.547421      85.850250
Somerset & Lockyer                       55.538462      97.353846
South West Coast VIC                     50.726257      79.979888
Southern Highlands & Tablelands          50.215054      79.918280
Sunshine Coast                           49.864865      86.640283
Sydney                                   71.882004     134.751041
Tamworth & North West NSW                61.278689     110.108197
Toowoomba & Darling Downs                51.076336      83.141221
Tumut, Snowy & Monaro                    67.558140     133.081395
Wagga Wagga & Riverina                   54.501247      87.087282
West Gippsland & Latrobe Valley          57.013889      93.586806
Western QLD                              69.357977     134.128405
Whyalla & Eyre Peninsula                 53.980100      86.000000
Wollongong, Illawarra & South Coast      55.829813      90.895594
Yarra Valley

# Top 5 cities that employees more well-paid (highest average salary)

In [70]:
salaryRangeLoc['avgSalary'] = (salaryRangeLoc['LowestSalary'] + salaryRangeLoc['HighestSalary'])/2
top5 = salaryRangeLoc['avgSalary'].nlargest(5)
top5

Location
ACT                                 142.446320
Port Macquarie & Mid North Coast    110.708955
Sydney                              103.316523
Western QLD                         101.743191
Geraldton, Gascoyne & Midwest       101.123167
Name: avgSalary, dtype: float64

In [71]:
patternPosting = data.groupby('Date').size()
patternPosting.head(40)

Date
2018-10-01T00:00:00.000Z     1023
2018-10-02T00:00:00.000Z     1980
2018-10-03T00:00:00.000Z     4945
2018-10-04T00:00:00.000Z     8506
2018-10-05T00:00:00.000Z     5711
2018-10-06T00:00:00.000Z      471
2018-10-07T00:00:00.000Z     1037
2018-10-08T00:00:00.000Z     4995
2018-10-09T00:00:00.000Z     6329
2018-10-10T00:00:00.000Z     8337
2018-10-11T00:00:00.000Z     1221
2018-10-12T00:00:00.000Z     1324
2018-10-13T00:00:00.000Z      196
2018-10-14T00:00:00.000Z      239
2018-10-15T00:00:00.000Z     5210
2018-10-16T00:00:00.000Z     6267
2018-10-17T00:00:00.000Z    13059
2018-10-18T00:00:00.000Z     8121
2018-10-19T00:00:00.000Z     5989
2018-10-20T00:00:00.000Z      422
2018-10-21T00:00:00.000Z     1145
2018-10-22T00:00:00.000Z      688
2018-10-23T00:00:00.000Z      827
2018-10-24T00:00:00.000Z      835
2018-10-25T00:00:00.000Z     1205
2018-10-26T00:00:00.000Z     1701
2018-10-27T00:00:00.000Z      232
2018-10-28T00:00:00.000Z      291
2018-10-29T00:00:00.000Z     4656
2018-10-3

Pattern of posting: The postings are not consistent at a particular time each month and are often posted at the middle of a month

## 3) Study the market by sectors (Vu)

In [72]:
data.head()

Title  \
0                           Casual Stock Replenisher   
1                           Casual Stock Replenisher   
2  RETAIL SALES SUPERSTARS and STYLISTS Wanted - ...   
3                              Team member - Belrose   
4  Business Banking Contact Centre Specialist, Ni...   

                                          Company                      Date  \
0                                     Aldi Stores  2018-10-07T00:00:00.000Z   
1                                     Aldi Stores  2018-10-07T00:00:00.000Z   
2                             LB Creative Pty Ltd  2018-10-07T00:00:00.000Z   
3                          Anaconda Group Pty Ltd  2018-10-07T00:00:00.000Z   
4  Commonwealth Bank - Business & Private Banking  2018-10-07T00:00:00.000Z   

                  Location                         Area  \
0                   Sydney  North West & Hills District   
1    Richmond & Hawkesbury                          NaN   
2                 Brisbane          CBD & Inner Suburbs   
3  Gosford & Central Coast                          NaN   
4                   Sydney        Ryde & Macquarie Park   

                   Classification  SubClassification  \
0      Retail & Consumer Products  Retail Assistants   
1      Retail & Consumer Products  Retail Assistants   
2      Retail & Consumer Products  Retail Assistants   
3      Retail & Consumer Products  Retail Assistants   
4  Call Centre & Customer Service    Sales - Inbound   

                                         Requirement FullDescription  \
0  Our Casual Stock Replenishers pride themselves...             NaN   
1  Our Casual Stock Replenishers pride themselves...             NaN   
2  BRAND NEW FLAGSHIP STORE OPENING - SUNSHINE PLAZA             NaN   
3  Bring it on - do you love the great outdoors a...             NaN   
4  We are seeking highly articulate, enthusiastic...             NaN   

   LowestSalary  HighestSalary JobType  
0             0             30     NaN  
1             0             30     NaN  
2             0             30     NaN  
3             0             30     NaN  
4             0             30     NaN

# Sectors with the highest market share

In [73]:
highestMarketShare = data.groupby('Classification').size()
highestMarketShare

Classification
Accounting                                 9368
Administration & Office Support           10020
Advertising, Arts & Media                   833
Banking & Financial Services               4633
CEO & General Management                    844
Call Centre & Customer Service             3971
Community Services & Development           4485
Construction                               8372
Consulting & Strategy                      1206
Design & Architecture                      1811
Education & Training                       9280
Engineering                                6458
Farming, Animals & Conservation             882
Government & Defence                       7443
Healthcare & Medical                      17085
Hospitality & Tourism                     15523
Human Resources & Recruitment              4144
Information & Communication Technology    22796
Insurance & Superannuation                 1474
Legal                                      4237
Manufacturing, Transport 

Sector with the highest market share: Information & Communication Technology (22796)

# Sub-Sectors which are the main spot-light in each sectors# 

In [74]:
subSec = data.groupby('Classification')['SubClassification'].max()
subSec

Classification
Accounting                                                             Treasury
Administration & Office Support           Records Management & Document Control
Advertising, Arts & Media                                            Promotions
Banking & Financial Services                                           Treasury
CEO & General Management                                                  Other
Call Centre & Customer Service                         Supervisors/Team Leaders
Community Services & Development               Volunteer Coordination & Support
Construction                                                          Surveying
Consulting & Strategy                                       Strategy & Planning
Design & Architecture                                  Web & Interaction Design
Education & Training                            Workplace Training & Assessment
Engineering                                           Water & Waste Engineering
Farming, Animals & Conser

# Salary range for each sector/sub-sector

In [75]:
salaryRangeSec = data.groupby('Classification')['LowestSalary', 'HighestSalary'].mean()
salaryRangeSec

LowestSalary  HighestSalary
Classification                                                     
Accounting                                 67.137062     108.139838
Administration & Office Support            40.057884      57.147705
Advertising, Arts & Media                  56.062425      84.644658
Banking & Financial Services               78.508526     143.533779
CEO & General Management                  126.741706     384.127962
Call Centre & Customer Service             39.790985      57.398136
Community Services & Development           47.977703      68.732219
Construction                               89.939083     163.196011
Consulting & Strategy                     101.948590     213.184909
Design & Architecture                      67.355052     111.046383
Education & Training                       59.173491      86.236530
Engineering                                90.346857     152.014710
Farming, Animals & Conservation            46.972789      68.621315
Government & Defence                       75.943840     131.460701
Healthcare & Medical                       72.934738     162.015920
Hospitality & Tourism                      33.081879      47.406558
Human Resources & Recruitment              69.734556     109.499276
Information & Communication Technology    108.765573     256.279084
Insurance & Superannuation                 67.876526     105.000000
Legal                                      87.481709     156.894501
Manufacturing, Transport & Logistics       46.964286      70.607708
Marketing & Communications                 67.825101     101.468750
Mining, Resources & Energy                 92.293257     164.978849
Real Estate & Property                     57.927504     100.794687
Retail & Consumer Products                 36.608631      53.476806
Sales                                      63.446237     101.293263
Science & Technology                       74.320442     127.893923
Self Employment                            70.967742     111.806452
Sport & Recreation                         45.203252      73.359756
Trades & Services                          44.829217      61.948821

## Compare salary range betweeen sectors


In [76]:
salaryRangeSec['AvgSalary'] = (salaryRangeSec['LowestSalary'] + salaryRangeSec['HighestSalary'])/2
salaryRangeSec['AvgSalary']

Classification
Accounting                                 87.638450
Administration & Office Support            48.602794
Advertising, Arts & Media                  70.353541
Banking & Financial Services              111.021153
CEO & General Management                  255.434834
Call Centre & Customer Service             48.594561
Community Services & Development           58.354961
Construction                              126.567547
Consulting & Strategy                     157.566750
Design & Architecture                      89.200718
Education & Training                       72.705011
Engineering                               121.180784
Farming, Animals & Conservation            57.797052
Government & Defence                      103.702271
Healthcare & Medical                      117.475329
Hospitality & Tourism                      40.244218
Human Resources & Recruitment              89.616916
Information & Communication Technology    182.522328
Insurance & Superannuation     

In [77]:
nameLowSec = salaryRangeSec['AvgSalary'].idxmin()
valueLowSec = salaryRangeSec['AvgSalary'].values.min()
print("Lowest Average Salary: ", nameLowSec, ":", valueLowSec)

Lowest Average Salary:  Hospitality & Tourism : 40.24421825678026


In [78]:
nameHighSec = salaryRangeSec['AvgSalary'].idxmax()
valueHighSec = salaryRangeSec['AvgSalary'].values.max()
print("Highest Average Salary: ", nameHighSec, ":", valueHighSec)

Highest Average Salary:  CEO & General Management : 255.43483412322274


Conclusion: As above, between all sectors:
    Lowest Salary: Hospitality and Tourism (33.081879),
    Highest Salary: CEO & General Management (384.127962).
About the average salary:
    Lowest Salary: Hospitality & Tourism (40.24421825678026),
    Highest Salary: CEO & General Management (255.43483412322274).

In [79]:
salaryRangeSubSec = data.groupby('SubClassification')['LowestSalary', 'HighestSalary'].mean()
salaryRangeSubSec

LowestSalary  HighestSalary
SubClassification                                                
Account & Relationship Management        69.433155     102.520856
Accounts Officers/Clerks                 44.090420      66.009688
Accounts Payable                         42.241055      61.745763
Accounts Receivable/Credit Control       44.019608      60.374183
Actuarial                                82.857143     108.392857
Administration                           38.216561      53.991507
Administrative Assistants                36.837270      53.259624
Aerospace Engineering                    98.627451     171.313725
Aged & Disability Support                41.856562      58.761445
Agency Account Management                60.952381      85.639456
Agronomy & Farm Services                 51.948052      70.129870
Air Conditioning & Refrigeration         52.543860      71.524561
Air Force                                85.000000     106.666667
Airlines                                 41.896552      55.344828
Ambulance/Paramedics                     68.367347     106.306122
Analysis & Reporting                     86.383648     143.591195
Analysts                                 92.270916     163.725100
Architects                              137.596899     427.099483
Architectural Drafting                   53.654822      77.756345
Architecture                             67.591743     104.543578
Army                                     84.000000     251.800000
Art Direction                            60.967742      79.354839
Assembly & Process Work                  34.121339      49.057531
Assessment                               54.264706      77.058824
Assistant Accountants                    43.957845      60.653396
Audit - External                         58.171429      80.394286
Audit - Internal                         91.034483     161.896552
Automotive Engineering                   60.250000     100.225000
Automotive Trades                        43.660331      60.151747
Aviation Services                        66.582278     121.854430
...                                            ...            ...
Teaching - Secondary                     61.231672      81.981427
Teaching - Tertiary                      91.414474     136.128289
Teaching - Vocational                    70.404218      96.409490
Teaching Aides & Special Needs           43.873239      57.183099
Team Leaders                            114.971098     196.872832
Team Leaders/Supervisors                 57.101449      78.515942
Technical Writing                       102.474227     244.185567
Technicians                              45.555556      61.679587
Telecommunications                       78.515406     124.450980
Testing & Quality Assurance             111.095066     242.819495
Tour Guides                              35.227273      50.000000
Trade Marketing                          59.545455      82.272727
Training & Development                   85.753425     137.500000
Travel Agents/Consultants                39.946903      53.876106
Treasury                                 85.176471     145.835294
Tutoring                                 64.000000      99.136842
Underwriting                             80.183486     141.422018
Urban Design & Planning                  79.021739     135.391304
Valuation                                70.952381     112.357143
Veterinary Services & Animal Welfare     37.577640      53.291925
Volunteer Coordination & Support         55.000000     109.952381
Waiting Staff                            27.824721      41.560405
Warehousing, Storage & Distribution      38.261826      55.931518
Water & Waste Engineering                84.305556     127.409722
Web & Interaction Design                101.270492     208.258197
Web Development & Production             89.004425     165.643805
Welders & Boilermakers                   48.017316      64.735931
Winery & Viticulture                     58.333333     105.633333
Workers' Com

# Compare the salary range between sub-sectors

In [80]:
salaryRangeSubSec['AvgSalary'] = (salaryRangeSubSec['LowestSalary'] + salaryRangeSubSec['HighestSalary'])/2
salaryRangeSubSec['AvgSalary']

SubClassification
Account & Relationship Management        85.977005
Accounts Officers/Clerks                 55.050054
Accounts Payable                         51.993409
Accounts Receivable/Credit Control       52.196895
Actuarial                                95.625000
Administration                           46.104034
Administrative Assistants                45.048447
Aerospace Engineering                   134.970588
Aged & Disability Support                50.309003
Agency Account Management                73.295918
Agronomy & Farm Services                 61.038961
Air Conditioning & Refrigeration         62.034211
Air Force                                95.833333
Airlines                                 48.620690
Ambulance/Paramedics                     87.336735
Analysis & Reporting                    114.987421
Analysts                                127.998008
Architects                              282.348191
Architectural Drafting                   65.705584
Architecture 

In [81]:
nameLowSubSec = salaryRangeSubSec['AvgSalary'].idxmin()
valueLowSubSec = salaryRangeSubSec['AvgSalary'].values.min()
print("Lowest Average Salary: ", nameLowSubSec, ":", valueLowSubSec)

Lowest Average Salary:  Waiting Staff : 34.69256284916201


In [82]:
nameHighSubSec = salaryRangeSubSec['AvgSalary'].idxmax()
valueHighSubSec = salaryRangeSubSec['AvgSalary'].values.max()
print("Highest Average Salary: ", nameHighSubSec, ":", valueHighSubSec)

Highest Average Salary:  General Practitioners : 412.2347585114806


Conclusion: As above, between all sub-sectors:
    Lowest Salary: Waiting Staff (27.824721),
    Highest Salary: General Practitioners (669.940618).
About the average salary:
    Lowest Salary: Waiting Staff (34.69256284916201),
    Highest Salary: General Practitioners (412.2347585114806).

# Trending of market:

In [83]:
highestMarketShare.nlargest(5)

Classification
Information & Communication Technology    22796
Trades & Services                         17976
Healthcare & Medical                      17085
Hospitality & Tourism                     15523
Manufacturing, Transport & Logistics      12376
dtype: int64

As above we can see those are the 5 most popular jobs as well as always require a huge amount of human resources.
So in order to guarantee a job in the future, I would suggest that student heading on studying IT or economy or both at the same time.

## 4) Visualize the result on an interactive web page

In [89]:
##import
from adjustText import adjust_text
from mpl_toolkits.basemap import basemap
from geopy.geocoder import Nominatim
from geopy.exc import GeocoderTimedOut

ModuleNotFoundError: No module named 'geopy'

## 5) Using PySpark for data analysis

## PART 3

## 1) The finding of your data analytics (Vu)

## 2) Actions for balancing the markets do you suggest based on your finding? (Vu)

## 3) Refine the data analytics?(Vu)

## 4) Are there any implications for employer and employees base on the finding you obtained? Justify your answer (Vu)

## 5) Present and visualize your data story on a online web page[optional]